docs/performance/inserts/bulk.rst

```diff
@@ -0,0 +1,419 @@
+    source file.
+
+
+.. _bulk-bulk-size:
+
+Bulk size
+---------
+
+You can improve on the example above by configuring the `bulk_size`_
+option, like so::
+
+    cr> COPY users FROM '/tmp/best_practice_data/users.json'
+    ... WITH (bulk_size = 2000);
+    COPY OK, 150 rows affected (... sec)
+
+The ``bulk_size`` option specifies the amount of lines that are read at once
+while importing. This option defaults to ``10000``.
+
+.. HIDE:
+
+    cr> REFRESH TABLE users;
+    REFRESH OK, 1 row affected (... sec)
+
+    cr> delete from users;
+    DELETE OK, 150 rows affected (... sec)
+
+    cr> REFRESH TABLE users;
+    REFRESH OK, 1 row affected (... sec)
+
+.. TIP::
+
+    In our example use-case, configuring ``bulk_size`` will not make any
+    practical difference.
+
+    However, if you have a more complex dataset with a lot of columns and large
+    values, it makes sense to decrease the ``bulk_size``.
+
+    A ``bulk_size`` setting that is too high might consume a lot of node
+    resources. A low ``bulk_size`` can increase the overhead resource
+    utilization per request.
+
+
+.. _bulk-compression:
+
+Compression
+-----------
+
+We recommend that you use `gzip`_ to compress your JSON files.
+
+However, CrateDB does not automatically detect file compression, so you will
+need to specify ``gzip`` compression, like so::
+
+    cr> COPY users FROM '/tmp/best_practice_data/users.json.gz'
+    ... WITH (compression = 'gzip');
+    COPY OK, 150 rows affected (... sec)
+
+.. HIDE:
+
+    cr> REFRESH TABLE users;
+    REFRESH OK, 1 row affected (... sec)
+
+
+.. _bulk-split-partitions:
+
+Split your tables into partitions
+=================================
+
+You can split your table into `partitions`_ to improve performance.
+
+.. HIDE:
+
+    cr> DROP TABLE users;
+    DROP OK, 1 row affected (... sec)
+
+Partitions can be created using the ``CREATE TABLE`` statement and a
+``PARTITIONED BY`` clause to specify a partition column. For example::
+
+    cr> CREATE TABLE users (
+    ...   id INT primary key,
+    ...   name TEXT,
+    ...   day_joined TIMESTAMP primary key,
+    ...   bio TEXT INDEX using fulltext,
+    ...   address OBJECT (dynamic) AS (
+    ...     city TEXT,
+    ...     country TEXT
+    ...   )
+    ... ) CLUSTERED INTO 6 shards
+    ... PARTITIONED BY (day_joined)
+    ... WITH (number_of_replicas = 0);
+    CREATE OK, 1 row affected (... sec)
+
+.. NOTE::
+
+    If a `primary key`_ was explicitly declared, the partition column has to be a
+    component of the primary key.
+
+A value identifying the target partition column *should* be defined in the
+``COPY FROM`` statement using the ``PARTITION`` clause::
+
+    cr> COPY users PARTITION (day_joined=1408312800)
+    ... FROM '/tmp/best_practice_data/users_1408312800.json';
+    COPY OK, 23 rows affected (... sec)
+
+When you use the ``PARTITION`` clause, CrateDB doesn't need to resolve the
+partition for each row that is imported, which aids performance.
```

Comment:
```suggestion
partition for each row that is imported, which also improves performance.
```

Reply:
disagree. the "also" is incorrect because the only reason we're mentioning the previous clause is *because* it has an impact on performance

additionally, I considered using "improves" here, but I shied away from it because I think it's too-definitive a statement to make. "may improve performance", on the other hand, sounds too meek

I settled on "aids" because it strikes the middle ground of communicating that it "helps" performance even if it practically makes no difference

I didn't use "helps" because I don't feel like "which helps performance" strictly good grammar in this context because there is a verb missing (i.e., helps performance do WHAT?)